modules/DOMUtils.js

```diff
@@ -17,7 +17,7 @@ export function removeEventListener(node, event, listener) {
 export function getHashPath() {
   // We can't use window.location.hash here because it's not
   // consistent across browsers - Firefox will pre-decode it!
-  return window.location.href.split('#')[1] || ''
+  return window.location.href.split('#').slice(1).join('#') || ''
```

Comment:
Use `indexOf` and `substr` to avoid extra work with arrays. It doesn't lose at all on readability.